spring-integration-test/src/main/java/org/springframework/integration/test/context/MockIntegrationContext.java

```diff
@@ -112,15 +120,31 @@ public void instead(String pollingAdapterId, MessageSource<?> mockMessageSource,
 		instead(pollingAdapterId, mockMessageSource, SourcePollingChannelAdapter.class, "source", autoStartup);
 	}
 
-	private void instead(String endpointId, Object mock, Class<?> endpointClass, String property,
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler) {
+		instead(consumerEndpointId, mockMessageHandler, true);
+	}
+
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler, boolean autoStartup) {
+		instead(consumerEndpointId, mockMessageHandler, IntegrationConsumer.class, "handler", autoStartup);
+		if (mockMessageHandler instanceof MessageProducer) {
```

Comment:
Well, yes, probably I've put the cart before the horse supposing that this solution should be somehow capable of dealing with other types of components such as Inbound Gateway. I think mocking of `MessageHandler`s will be more in-demand than other components including Inbound Gateways.